config/src/main/java/io/scalecube/config/JsonDocumentConfigPropertyImpl.java

```diff
@@ -0,0 +1,80 @@
+package io.scalecube.config;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectReader;
+import io.scalecube.config.source.LoadedConfigProperty;
+import io.scalecube.config.utils.ObjectMapperHolder;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of {@link ObjectConfigProperty}.
+ *
+ * @param <T> type of the property value
+ */
+class JsonDocumentConfigPropertyImpl<T> extends AbstractConfigProperty<T>
+    implements ObjectConfigProperty<T> {
+
+  protected volatile T value;
```

Comment:
I'm not sure that we need this field at all.

Reply:
we are testing this value against the previous one in order to trigger changes only when a real change was made, e.g:
Someone changed the document to be _same_ document by value. Should we trigger a change? the answer is **NO**. but actually, as we read the value from the configuration service itself we do not know if the document was accessed or not (the value is not _changed_ - we always load it)

Reply:
1) What does `configRegistry` do when it parses incorrect integer to? It throws number format exception.
2) our `configRegistry` received the new value of the JSON. we invoke the callback that says that we have changed value - and it's not our problem that value is incorrect and we can't parse it into necessary type, we just should throw an exception. We can read this property as a string and it will be a valid case as well.

Reply:
Changed the field on superclass to be protected